src/main/java/org/jabref/gui/undo/CountingUndoManager.java

```diff
@@ -27,16 +27,20 @@ public synchronized boolean addEdit(UndoableEdit edit) {
 
     @Override
     public synchronized void undo() throws CannotUndoException {
-        super.undo();
-        current--;
-        postUndoRedoEvent();
+        if (canUndo()) {
+            super.undo();
+            current--;
+            postUndoRedoEvent();
+        }
```

Comment:
I think, the exception should be caught by the caller. On the one hand, JabRef should not enable Ctrl+Z if it is not possible to do undo something. On the other hand, we will surely eventually change all the undo/redo things. Anyone already said Redux/MobX / Command Pattern everywhere?